include/seqan3/alignment/cigar_conversion/alignment_from_cigar.hpp

```diff
@@ -0,0 +1,226 @@
+ * `AC-GA` wehre `-` represents a gap.
+ * In the CIGAR string the gap in the query/read is represented by `1D`.
+ *
+ * The full alignment consist of two aligned sequences (read and reference).
+ * In the above example, the alignment
+ * ```
+ * position   01234
+ * reference  ACTGA
+ * read       AC-GA
+ * ```
+ * is represented by a tuple of the aligned reference at the first position (std::get<0>) and the aligned read at the
+ * second position (std::get<1>): `(ACTGA,AC-GA)`.
+ *
+ * ### IO Example
+ *
+ * A more realistic example is that you get the information directly from a SAM file:
+ *
+ * \include test/snippet/alignment/cigar_conversion/alignment_from_cigar_io.cpp
+ *
+ * \sa seqan3::sam_file_input
+ * \sa seqan3::cigar
+ */
+template <typename reference_type, typename sequence_type>
+inline auto alignment_from_cigar(std::vector<cigar> const & cigar_vector,
+                                 reference_type const & reference,
+                                 uint32_t const zero_based_reference_start_position,
+                                 sequence_type const & query)
+{
+    if (cigar_vector.empty())
+        throw std::logic_error{"An empty CIGAR is not a valid alignment representation."};
+
+    // compute the length of the aligned region in the reference sequence
+    // -------------------------------------------------------------------------
+    // this requires a first stream over the cigar vector.
+    uint32_t reference_length{0};
+    uint32_t query_length{0};
+
+    for (auto [cigar_count, cigar_operation] : cigar_vector)
+    {
+        if (('M'_cigar_operation == cigar_operation) || ('='_cigar_operation == cigar_operation)
+            || ('X'_cigar_operation == cigar_operation))
+        {
+            reference_length += cigar_count;
+            query_length += cigar_count;
+        }
+        else if ('D'_cigar_operation == cigar_operation)
+        {
+            reference_length += cigar_count;
+        }
+        else if ('I'_cigar_operation == cigar_operation)
+        {
+            query_length += cigar_count;
+        }
+    }
+
+    if (static_cast<size_t>(zero_based_reference_start_position + reference_length) > std::ranges::size(reference))
+        throw std::logic_error{"The CIGAR string indicates a reference length of at least "
+                               + std::to_string(zero_based_reference_start_position + reference_length)
+                               + " but the supplied reference sequence is only of size"
+                               + std::to_string(std::ranges::size(reference))};
+
+    // Get soft clipping at the start and end of the cigar string
+    // -------------------------------------------------------------------------
+    uint32_t soft_clipping_start{0};
+    uint32_t soft_clipping_end{0};
+
+    // Checks if the given index in the cigar vector is a soft clip.
+    auto soft_clipping_at = [&](size_t const index)
```

Comment:
```suggestion
    auto soft_clipping_at = [&cigar_vector](size_t const index)
```

Not sure about style. Doesn't affect performance, so no strong feelings.